### PR TITLE
Post350

### PR DESCRIPTION
**Pull request by TimMangan**
This PR contains PSFLauncher the scripting changes to support arguments including the %MsixPackageRoot% pseudo variable so that the scripts can reference package files as arguments to the script.  

It also has edge-case fixes for a few FileRedirectionFixup intercepts that improve application compatibility.

### CODE REVIEW

**Comment by TimMangan** [comment]
I'm not sure what the conflicts are all about.  If necessary I'll cancel this PR and rebuild from a fresh fork, but hoping that isn't necessary.

File: PsfLauncher/PsfPowershellScriptRunner.h [hunk 1]
```diff
@@ -56,11 +59,28 @@ class PsfPowershellScriptRunner
 	//RunStartingScript should return an error only if stopOnScriptError is true
 	void RunStartingScript()
 	{
+		LogString("StartingScript commandString",this->startingScriptInformation.commandString.c_str());
+		LogString("StartingScript currentDirectory", this->startingScriptInformation.currentDirectory.c_str());
+		if (this->startingScriptInformation.runInVirtualEnvironment)
```
Comment (dhoehna) [inline PsfLauncher/PsfPowershellScriptRunner.h]: Please use { and } even if the blocks are one line.

File: PsfLauncher/PsfPowershellScriptRunner.h [hunk 2]
```diff
@@ -139,11 +162,86 @@ class PsfPowershellScriptRunner
...
+				}
+				if (inString[0] == L'`' && inString[1] == L'\"' && inString[inString.length() - 2] == L'`' && inString[inString.length() - 1] == L'\"')
+				{
+					return inString.substr(2, inString.length() - 4);
+				}
+			}
+		}
+		return inString;
+	}
+	std::wstring EscapeFilename4PowerShell(std::filesystem::path inputPath)
```
Comment (dhoehna) [inline PsfLauncher/PsfPowershellScriptRunner.h]: Please replace 4 with Four.

File: PsfLauncher/main.cpp [hunk 3]
```diff
@@ -101,6 +147,105 @@ catch (...)
     return win32_from_caught_exception();
 }
 
+bool LaunchViaCreateProcessInContainer(std::wstring app, std::wstring fullCmd, std::wstring dir, bool cmdShow)
```
Comment (dhoehna) [inline PsfLauncher/main.cpp]: There is already a fix in PSF that I added last week to make sure Powershell.exe is always launched inside the container.  In StartProcessHelper.h there is a method called MakeAttributeList. This modifies the breakaway behavior.

I think you were seeing an issue that I recently fixed where the breakaway behavior was not correct in release builds. That issue has been fixed.  Please bring in all changes from develop to see them.

**Comment by dhoehna** [comment]
Hey Tim. All the merge conflicts might be because this branch does not have all the changes from the past two PR's I accepted.  You might want to either
1. Pull all changes from develop into this branch.
2. Make an entirly new branch that is current with develop and re-add your changes.

**Comment by TimMangan** [comment]
I'll rebalance, address other items above, and retest.  Will post back when done.

File: PsfLauncher/PsfPowershellScriptRunner.h (hunk 2 above)
Comment (TimMangan) [reply PsfLauncher/PsfPowershellScriptRunner.h]: ?? Not sure what the request here is. Are you suggesting to add a variable called "Four" with value 4 to replace use of 4 in the code around this area?

Comment (dhoehna) [reply PsfLauncher/PsfPowershellScriptRunner.h]: Oops.  I meant to replace 4 with For, not Four.

The varible name should be ExcapeFilenameForPowerShell.

**Comment by TimMangan** [comment]
Now that makes more sense!

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by TimMangan** [comment]
Upon request, I have resynched to the develop branch and have a replacement PR for this one.

**Pull request closed without merging**